tmdl_tool.cs

```diff
@@ -9,34 +9,50 @@ internal class tmdl_tool
         private const int ERROR_INVALID_ACTION = 0x667;
         private const int ERROR_PATH_NOT_FOUND = 0x3;
         private const int ERROR_NETWORK_ACCESS = 0x44;
+        private const int ERROR_UNEXPECTED_ERROR = 0x999;
+
+        private static string VersionString = "";
+
+        private static string workspaceXMLA = "";
+        private static string datasetName = "";
+        private static string tmdlfolderPath = "";
+        private static string action = "";
+        private static string settingsFilePath = "";
+        private static string appId = "";
+        private static string appSecret = "";
+        private static string tenantId = "";
+        private static string accessToken = "";
+        private static bool verbose = true;
 
         static void Main(string[] args)
         {
-            string workspaceXMLA;
-            string datasetName;
-            string tmdlfolderPath;
-            string action;
-            string settingsFilePath;
-            string appId;
-            string appSecret;
-            string tenantId;
-
-            Environment.ExitCode = ERROR_SUCCESS;
+
+            // Setting tmdl_tool.VersionString
+            var assemblyInfo = System.Reflection.Assembly.GetExecutingAssembly().GetName();
+            var tmdlLib = System.Reflection.Assembly.GetAssembly(typeof(Server));
+            var libName = tmdlLib?.GetName()?.Name ?? "Unknown";
+            var libVersion = tmdlLib?.GetName()?.Version?.ToString() ?? "Unknown";
+            VersionString = $"{assemblyInfo.Name} v.{assemblyInfo.Version}, {libName}: {libVersion}";
+
+            Environment.ExitCode = ERROR_UNEXPECTED_ERROR;
+
             PrintHelpIfRequested(args);
 
-            GetArguments(args, out workspaceXMLA, out datasetName, out tmdlfolderPath, out action, out settingsFilePath, out appId, out appSecret, out tenantId);
+            GetArguments(args);
+
+            GetSettings(settingsFilePath);

```

Comment:
Probably should switch the positions, to give arguments preference over settings.json?